### PR TITLE
German Dictionary for Hunspell

### PR DESCRIPTION
**Pull request by timor**
###### Motivation for this change
Add german dictionary to hunspell.

###### Things done
Building these dicts needs ispell, so I re-used the PR #3320 for that.
For the actual dict, used the code from this: #32777

Tested by building hunspell with german and en-US dictionaries on 17.09 and master and running `hunspell -D`, which then shows the german dictionary.  Also tested actual spelling correction, which gives the same result as the online spell checker (https://j3e.de/cgi-bin/spellchecker)

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by timor** [comment]
This PR would introduce ispell as a top-level package although it is only used (not even the latest version) for building the dictionary.  Is that acceptable?  Should ispell be only a local derivation in dictionaries.nix?  If so, what is the best way to do that?

**Comment by timor** [comment]
cc @Fuuzetsu

**Review by Fuuzetsu — APPROVED** [review]
It's ok with me the way it is in this PR.

**Comment by globin** [comment]
@GrahamcOfBorg build hunspellDicts.de-de ispell

File: pkgs/tools/text/ispell/default.nix [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    #sed -i -e "s|#define bzero.*||" -e "s|#define bcopy.*||" config.X
+
+    #XXX: getline is already defined (name clash), so rename it
+    #sed -i -e "s|getline|getlineXXX|g" correct.c
+
+    #XXX: strcmp is a macro, so importing it as function won't work
+    #sed -i -e 's| *extern int\s*strcmp.*||' ijoin.c
+
+    #XXX: buildhash uses int to store pointers, which doesn't work on 64bit
+    #sed -i -e 's|\s*int\s*strptr|intptr_t strptr|' -e 's|\s*int\s*x|intptr_t x|' buildhash.c
```
Comment (globin) [inline pkgs/tools/text/ispell/default.nix]: Should these be removed?

Comment (timor) [reply pkgs/tools/text/ispell/default.nix]: I removed them.

File: pkgs/tools/text/ispell/default.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
+{ stdenv, fetchurl, bison, ncurses }:
+
+stdenv.mkDerivation rec {
+  name = "ispell-3.3.02";
```
Comment (jtojnar) [inline pkgs/tools/text/ispell/default.nix]: This is outdated, current version is 3.4.00: https://www.cs.hmc.edu/~geoff/ispell.html

Comment (timor) [reply pkgs/tools/text/ispell/default.nix]: Yes.  The reason why I did not try to use the newest one is that ispell here is only used as a build dependency for the dictionary.  So I basically stuck to the version that someone else deemed working.

**Pull request merged**